配列を昇順に並び替える選択ソートプログラム作成

morikaku-ts opened the pull request.

Commit:
配列を昇順に並び替える選択ソートプログラム作成

Commit:
選択ソートプログラムの起動方法を記載

morikaku-ts committed:
昇順、降順に並び替える処理へ変更

morikaku-ts committed:
sorging_methodに無効な値が渡された場合の処理を追加

kyakumoto-ts commented on select_sort.rb:
```diff
@@ -1,23 +1,33 @@
+end
+
+def ascending_order(numbers, index)
+  exchange_number_index = numbers.index(numbers[index...numbers.length].min)
```
##### L23
`Array#min` を自分で書いてみてください。

kyakumoto-ts commented on select_sort.rb:
```diff
@@ -1,23 +1,33 @@
+end
+
+def descending_order(numbers, index)
+  exchange_number_index = numbers.index(numbers[index...numbers.length].max)
```
##### L28
`Array#max` を自分で書いてみてください。

kyakumoto-ts reviewed (commented):
すみません、見落としていました。
挑戦してみてください。

kyakumoto-ts commented on select_sort.rb:
```diff
@@ -0,0 +1,39 @@
+def descending_order(numbers, index)
+  exchange_number_index = numbers.index(numbers[index...numbers.length].max)
+  swap(numbers, index, exchange_number_index)
+end
```
##### L27 - L35
実行しなくても良い場合でも、必ず `swap` が呼ばれることになっています。
`swap` しなくて良い場合は、呼び出さない様にしましょう。

morikaku-ts committed:
配列内の最小値 / 最大値の検索方法をmin, maxメソッドを使用しない方法に変更

- sorting_methodに渡す引数がasc, desc以外の場合の中断処理を変更
- swapメソッドは、基準のindexと交換対象のindexが同一の場合には交換処理をしないよう変更
